ballerina/schema-client.bal

```diff
@@ -0,0 +1,320 @@
+            return error sql:NoRowsError("The selected table does not exist or the user does not have the required privilege level to view the table.");
+        } else if 'table is sql:Error {
+            return 'table;
+        } else {
+            TableDefinition tableDef = {
+                name: tableName,
+                'type: <sql:TableType>'table["table_type"]
+            };
+
+            if !(include == sql:NO_COLUMNS) {
+                TableDefinition|sql:Error tableDefError = self.addColumns(tableName, tableDef);
+        
+                if tableDefError is TableDefinition {
+                    tableDef = tableDefError;
+                }
+
+                if include == sql:COLUMNS_WITH_CONSTRAINTS {
+                    tableDefError = self.getConstraints(tableName, tableDef);
+
+                    if tableDefError is TableDefinition {
+                        tableDef = tableDefError;
+                    }
+                }    
+            }
+
+            return tableDef;
+        }
+    }
+
+    # Retrieves all routines in the database.
+    #
+    # + return - A string array containing the names of the routines or an `sql:Error`
+    isolated remote function listRoutines() returns string[]|sql:Error {
+        string[] routines = [];
+        stream<record {}, sql:Error?> results = self.dbClient->query(
+            `SELECT ROUTINE_NAME FROM INFORMATION_SCHEMA.ROUTINES
+            WHERE ROUTINE_CATALOG = ${self.database} AND ROUTINE_SCHEMA NOT IN ('pg_catalog', 'information_schema');`
+        );
+
+        do {
+            routines = check from record {} result in results
+                select <string>result["routine_name"];
+        } on fail error e {
+            return error(string `Error while listing routines in the ${self.database} database.`, cause = e);
+        }
+
+        check results.close();
+
+        return routines;
+    }
+
+    # Retrieves information relevant to the provided routine in the database.
+    #
+    # + name - The name of the routine
+    # + return - An 'sql:RoutineDefinition' with the relevant routine information or an `sql:Error`
+    isolated remote function getRoutineInfo(string name) returns sql:RoutineDefinition|sql:Error {
+        sql:RoutineDefinition|sql:Error routine = self.dbClient->queryRow(
+            `SELECT ROUTINE_TYPE, DATA_TYPE FROM INFORMATION_SCHEMA.ROUTINES 
+             WHERE ROUTINE_NAME = ${name};`
+        );
+
+        if routine is sql:NoRowsError {
+            return error sql:NoRowsError("Selected routine does not exist in the database, or the user does not have required privilege level to view it.");
+        } else if routine is sql:Error {
+            return routine;
+        } else {
+
+            sql:RoutineDefinition|sql:Error routineError = self.getParameters(name, routine);
+
+            if routineError is sql:RoutineDefinition {
+                routine = routineError;
+            }
+
+            return routine;
```

Comment:
```suggestion
            if routineError is sql:RoutineDefinition {
                routine = routineError;
            }

            return routine;
```

In this case, if `getParameters()` returns an error, there would be no way of identifying it.